### PR TITLE
Balance calc numbers

### PR DESCRIPTION
**Pull request by CyberiaResurrection**
Enforces the totals-related invariants mentioned in #16.

### CODE REVIEW

File: PyRoute/Calculation/TradeCalculation.py [hunk 1]
```diff
@@ -66,6 +72,17 @@ def __init__(self, galaxy, min_btn=13, route_btn=8, route_reuse=10, debug_flag=F
         # Are debugging gubbins turned on?
         self.debug_flag = debug_flag
 
+        self.shortest_path_tree = None
+        # Track inter-sector passenger imbalances
+        self.sector_passenger_balance = TradeBalance(stat_field="passengers", region=galaxy)
+        # Track inter-sector trade imbalances
+        self.sector_trade_balance = TradeBalance(stat_field="tradeExt", region=galaxy, target="trade")
+        # Track inter-allegiance passenger imbalances
+        self.allegiance_passenger_balance = TradeBalance(stat_field="passengers", region="galaxy", field="allegiances", star_field="allegiance_base")
```
Comment (tjoneslo) [inline PyRoute/Calculation/TradeCalculation.py:81]: Why is this region name quoted when none of the rest of them are?

File: PyRoute/Calculation/TradeCalculation.py [hunk 2]
```diff
@@ -409,9 +456,50 @@ def route_weight(self, star, target):
...
 
+
     def unilateral_filter(self, star):
         if star.zone in ['R', 'F']:
             return True
         if star.tradeCode.barren:
             return True
         return False
+
+    def is_sector_trade_balanced(self):
```
Comment (tjoneslo) [inline PyRoute/Calculation/TradeCalculation.py:467]: Shouldn't there also be an `is_allegiance_trade_balanced()` and `is_allegiance_pass_balanced()` methods?

File: PyRoute/StatCalculation.py [hunk 3]
```diff
@@ -208,6 +217,10 @@ def calculate_statistics(self, ally_match):
             for uwpStats in uwpName.values():
                 self.per_capita(None, uwpStats)
 
+        self.galaxy.trade.is_sector_trade_balanced()
+        self.galaxy.trade.is_sector_pass_balanced()
```
Comment (tjoneslo) [inline PyRoute/StatCalculation.py:221]: This is were we should call the `is_allegiance_trade_balanced()` methods?

**Review by tjoneslo — CHANGES_REQUESTED** [review]
There are some missing pieces here. I think.

File: PyRoute/Calculation/TradeCalculation.py (hunk 2 above)
Comment (CyberiaResurrection) [reply PyRoute/Calculation/TradeCalculation.py:467]: By symmetry, yes

File: PyRoute/StatCalculation.py (hunk 3 above)
Comment (CyberiaResurrection) [reply PyRoute/StatCalculation.py:221]: Yes-  lemme bolt them in

**Comment by CyberiaResurrection** [comment]
@tjoneslo , that should address your comments

**Review by tjoneslo — APPROVED** [review]

**Pull request merged**